### DISCUSSION TITLE
Can  add a TryGetValue overload with delegate parameters to the dictionary?

### DISCUSSION THREAD

**Discussion by zms9110750** [dotnet/runtime#69554] [General]
my extension method:
```csharp
public static V TryGetValue<K, V>(this Dictionary<K, V> dic, K key, Func<K, V> getValue)
{
	return dic.TryGetValue(key, out var value) ? value : dic[key] = getValue(key);
}
```
I wrote ugly. Maybe you'll give a better answer

**Comment by svick** [comment]
This is a duplicate of https://github.com/dotnet/runtime/issues/15059.

Though it seems you're more interested in convenience than in performance, in which case the resolution of only adding the low-level efficient `CollectionsMarshal.GetValueRefOrAddDefault` may be unsatisfying for you.
